### ISSUE TITLE
Connection Failed with Teamcity 2017.1.1

### ISSUE THREAD

**Issue by stallion2076a** [parautenbach/TeamCity-HipChat-Notifier#79]
Helllo ,

I am unable to connect my teamcity to Hichat , it says connection failed 

![image](https://cloud.githubusercontent.com/assets/5584645/26139850/b59ecbf0-3b16-11e7-94b6-ffb0eff0352c.png)

Just wondering if I got the URL right 

![image](https://cloud.githubusercontent.com/assets/5584645/26139876/e5de3594-3b16-11e7-8bbb-86c9e13151bd.png)

**Comment by andyofengland** [comment]
Did you ever get a resolution to this at all?

**Comment by BenedekFarkas** [comment]
The URL (in case you're using HipChat Cloud) should be `https://api.hipchat.com/v2/`.

**Comment by andyofengland** [comment]
I've tried that but I keep getting "Connection Failed".  Are there any specific firewall exceptions that are needed from the server?  Or is it just server ==> hipchat on 80?

**Comment by BenedekFarkas** [comment]
I don't think so. Btw the token needed for the connection needs to be assigned to a user, since account-wide tokens don't work with the v2 API (only v1). You need to set up a user (or use an existing one, like your own, if you're an admin, but I have a separate one) for sending notifications, then log in with it and go to `https://yourdomain.hipchat.com/account/api` and create an API token with the `Send notification` permission level.

**Comment by khaljimbo** [comment]
Have you been able to get this working? I've tried everything including the ideas suggest on #52 and #61 and it's still not working.

**Comment by webmastir** [comment]
@khaljimbo  I had a similar issue not too long ago. It ended up being the result of an expired access token. 
I highly doubt this is your problem, but it might be worth checking.

**Comment by BenedekFarkas** [comment]
@khaljimbo works for me with the token generated as described in my previous comment.

**Comment by khaljimbo** [comment]
I still can't get it to connect, I've created 3 different tokens. 

One with every permission
One with only the 3 in the last comment
One with everything from the hipchat administrator.

I still can't seem to get it to connect. Really want to use this enable our CI journey even better.

**Comment by stallion2076a** [comment]
It was the url that was wrong , after I fixed  it it worked fine

**Comment by khaljimbo** [comment]
@stallion2076a can you share your URL? 

I am using https://api.hipchat.com/v2 

I am using the Octopus Deploy hipchat notifier too which uses the same api url and that is working fine. I am also using TeamCity 2017.1.2

**Comment by khaljimbo** [comment]
So I was able to get it to connect by adding /rooms onto the end of the URL. I get a connection successful message. However it is not showing any rooms in the drop down when I click to choose them, the list just stays blank even after reloading and refreshing the page. 

I've got it completely working now. All I needed was the / on the end on v2.

**Comment by webmastir** [comment]
@khaljimbo Ah! Glad it's fixed. I had to do the same thing as well.

**Comment by flocsy** [comment]
I can't make it work. I tried it both with https://api.hipchat.com/v2, https://api.hipchat.com/v2/rooms,  https://mydomain.hipchat.com/v2/, https://mydomain.hipchat.com/v2/rooms. My TC version is: 2017.2.1 (build 50732)

**Comment by BenedekFarkas** [comment]
@flocsy working for me on the same TC version. The URL should be `https://api.hipchat.com/v2/`. Have you got the token set up correctly (as discussed above)? If you get a "Connection successful", you still need to set it up for your TC projects.

**Comment by flocsy** [comment]
URL: https://api.hipchat.com/v2/, token from https://mysubdomain.hipchat.com/account/api (Send Notification scope with my user) and I get "test connection" (tried before and after save) just gives me "Connection failed". I have no idea what's the problem, there's no indication at all.

When I click on the button it starts an ajax call to (fake host, token :)

http://my-tc.example.com:8901/configureHipChat.html?test=1&apiUrl=https%3A%2F%2Fapi.hipchat.com%2Fv2%2F&apiToken=abcdeflwef3efg23ef23ewfvfdsad23wesdf3w3wer32

That gives a 400 error

**Comment by BenedekFarkas** [comment]
I just realized that my token has both `Send Notification` and `View Group` scope. If you only have `View Group`, then the connection test will be successful, but it won't be able to send notifications.

So you need both of those scopes for connection test to be successful and be able to assign HipChat rooms to TeamCity projects and then actually send notifications.

FYI it will only be able to access your public rooms and private ones that the user (to whom the token belongs) is invited to.

EDIT: It's actually in the readme. oops

**Comment by flocsy** [comment]
na így már működik!

**Comment by BenedekFarkas** [comment]
Remek! :D

**Comment by Funbit** [comment]
OMG, I've spent almost an hour trying to find out why connection is failed.
@BenedekFarkas 's answer is correct, both scopes are required for connection test to be successful! Thanks!
